Modify max_instances to take API's result as default value (#7153)
GoogleCloudPlatform/magic-modules: mmv1/third_party/terraform/website/docs/d/cloudfunctions_function.html.markdown

* `service_account_email` - The service account email to be assumed by the cloud function.
* `vpc_connector` - The VPC Network Connector that this cloud function can connect to. 
* `vpc_connector_egress_settings` - The egress settings for the connector, controlling what traffic is diverted through it.

> * `max_instances` - The limit on the maximum number of function instances that may coexist at a given time. This is optional field. If value is not provided by user, a default value will be assigned by cloud function automatically. 

Reviewer: ```suggestion
* `max_instances` - The limit on the maximum number of function instances that may coexist at a given time. If unset or set to `0`, the API default will be used. 
```

It's important to mention that users can unset this field by setting it to 0, because Optional + Computed fields don't register as "changed" if you unset them.
